mosspy/download_report.py

```diff
@@ -38,9 +38,10 @@ def process_url(url, urls, base_url, path):
 
             if link not in urls:                    
                 urls.append(link)
-
-    f = open(path + file_name, 'w')
-    f.write(str(soup)) # saving soup will save updated href
+    
+    f = open(os.path.join(path, file_name), 'wb')
+    html_encoding = soup.original_encoding
+    f.write(bytes(soup.decode(html_encoding), encoding=html_encoding))
```

Comment:
@nedchu You are decoding and then encoding again, why not encode once `f.write(soup.encode(soup.original_encoding))`. Secondly, using bytes will break on Python 2.